tests/integration/local/start_api/test_start_api_with_terraform_application.py

```diff
@@ -99,6 +108,89 @@ def test_successful_request(self):
         self.assertEqual(response.json(), {"message": "hello world"})
 
 
+@skipIf(
+    not CI_OVERRIDE,
+    "Skip Terraform test cases unless running in CI",
+)
+@pytest.mark.flaky(reruns=3)
+@parameterized_class(
+    [
+        {
+            "terraform_application": "lambda-auth-openapi",
+            "expected_error_message": "Error: AWS SAM CLI is unable to process a Terraform project that uses an OpenAPI"
+            " specification to define the API Gateway resource.",
+        },
+        {
+            "terraform_application": "terraform-api-simple-multiple-resources-limitation",
+            "expected_error_message": "Error: AWS SAM CLI could not process a Terraform project that contains a source "
+            "resource that is linked to more than one destination resource.",
+        },
+        {
+            "terraform_application": "terraform-api-simple-local-variables-limitation",
+            "expected_error_message": "Error: AWS SAM CLI could not process a Terraform project that uses local "
+            "variables to define linked resources.",
+        },
+    ]
+)
+class TestStartApiTerraformApplicationLimitations(TerraformStartApiIntegrationBase):
+    @classmethod
+    def setUpClass(cls):
```

Comment:
question: how come we are setting up the class here again?

Reply:
Sorry I did not get your question. I am overriding the setupClass function defined in parent class. This will be called instead of the one in the parent class

Reply:
I do not want to call 'sam local start-api' in the setup class, as it is supposed to fail. So I override this method to skip executing sam local command

Reply:
That makes sense, since the parent will fail before it ever gets to the test, thanks!